applications/vanilla/controllers/class.vanillasettingscontroller.php

```diff
@@ -716,6 +716,37 @@ public function manageCategories() {
         $this->render();
     }
 
+    /**
+     * Move a category to a different parent.
+     *
+     * @param int $categoryID Unique ID for the category to move.
+     * @throws Exception if category is not found.
+     */
+    public function movecategory($categoryID) {
+        $category = CategoryModel::categories($categoryID);
+
+        if (!$category) {
+            throw notFoundException();
+        }
+
+        $this->Form->setModel($this->CategoryModel);
+        $this->Form->addHidden('CategoryID', $categoryID);
+        $this->setData('Category', $category);
+
+        $parentCategories = CategoryModel::getAncestors($categoryID);
+        array_pop($parentCategories);
+        if (!empty($parentCategories)) {
+            $this->setData('ParentCategories', array_column($parentCategories, 'Name', 'CategoryID'));
+        }
+
+        if ($this->Form->authenticatedPostBack()) {
+            $this->Form->save();
```

Comment:
Keep in mind that because of Vanilla's very permissive data handling, you're enabling **any** category data to be set via this endpoint currently. Might consider whitelisting `CategoryID` and `ParentCategoryID`.